From 884bfce8e77eff054ee553c4cfb64288b5aec970 Mon Sep 17 00:00:00 2001
From: Alejandro Sierra <b18039@freescale.com>
Date: Mon, 6 Jun 2016 18:28:51 -0500
Subject: [PATCH 21/23] ARM: dts: enable Murata's bt on imx6sxscm evb

Enabla Murata's bluetooth on imx6sxscm evb, along with Wifi.
Use a single dts file to enable both wifi and bluetooth.

Signed-off-by: Alejandro Sierra <b18039@freescale.com>
---
 arch/arm/boot/dts/Makefile                         |   6 +-
 arch/arm/boot/dts/imx6sxscm-1gb-evb-btwifi-ldo.dts |  10 ++
 arch/arm/boot/dts/imx6sxscm-1gb-evb-wifi-ldo.dts   |  36 ------
 .../boot/dts/imx6sxscm-512mb-evb-btwifi-ldo.dts    |  10 ++
 arch/arm/boot/dts/imx6sxscm-512mb-evb-wifi-ldo.dts |  34 ------
 .../arm/boot/dts/imx6sxscm-epop-evb-btwifi-ldo.dts |   9 ++
 arch/arm/boot/dts/imx6sxscm-epop-evb-wifi-ldo.dts  |   9 --
 arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi        | 127 +++++++++++++++++++++
 arch/arm/boot/dts/imx6sxscm-evb.dtsi               |   7 ++
 9 files changed, 166 insertions(+), 82 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6sxscm-1gb-evb-btwifi-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6sxscm-1gb-evb-wifi-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6sxscm-512mb-evb-btwifi-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6sxscm-512mb-evb-wifi-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6sxscm-epop-evb-btwifi-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6sxscm-epop-evb-wifi-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 74efe92..9a2c6eb 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -247,19 +247,19 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx6sxscm-512mb-evb-m4-ldo.dtb \
 	imx6sxscm-512mb-evb-mqs-ldo.dtb \
 	imx6sxscm-512mb-evb-sai-ldo.dtb \
-	imx6sxscm-512mb-evb-wifi-ldo.dtb \
+	imx6sxscm-512mb-evb-btwifi-ldo.dtb \
 	imx6sxscm-1gb-evb-ldo.dtb \
 	imx6sxscm-1gb-evb-lcdif1-ldo.dtb \
 	imx6sxscm-1gb-evb-m4-ldo.dtb \
 	imx6sxscm-1gb-evb-mqs-ldo.dtb \
 	imx6sxscm-1gb-evb-sai-ldo.dtb \
-	imx6sxscm-1gb-evb-wifi-ldo.dtb \
+	imx6sxscm-1gb-evb-btwifi-ldo.dtb \
 	imx6sxscm-epop-evb-ldo.dtb \
 	imx6sxscm-epop-evb-lcdif1-ldo.dtb \
 	imx6sxscm-epop-evb-m4-ldo.dtb \
 	imx6sxscm-epop-evb-mqs-ldo.dtb \
 	imx6sxscm-epop-evb-sai-ldo.dtb \
-	imx6sxscm-epop-evb-wifi-ldo.dtb \
+	imx6sxscm-epop-evb-btwifi-ldo.dtb \
 	vf610-twr.dtb \
 	imx7d-12x12-lpddr3-arm2.dtb \
 	imx7d-12x12-ddr3-arm2.dtb \
diff --git a/arch/arm/boot/dts/imx6sxscm-1gb-evb-btwifi-ldo.dts b/arch/arm/boot/dts/imx6sxscm-1gb-evb-btwifi-ldo.dts
new file mode 100644
index 0000000..a31c4d6
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-1gb-evb-btwifi-ldo.dts
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sxscm-1gb-evb-ldo.dts"
+#include "imx6sxscm-evb-btwifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6sxscm-1gb-evb-wifi-ldo.dts b/arch/arm/boot/dts/imx6sxscm-1gb-evb-wifi-ldo.dts
deleted file mode 100644
index 1658385..0000000
--- a/arch/arm/boot/dts/imx6sxscm-1gb-evb-wifi-ldo.dts
+++ /dev/null
@@ -1,36 +0,0 @@
-/*
- * Copyright (C) 2016 Freescale Semiconductor, Inc.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-
-#include "imx6sxscm-1gb-evb-ldo.dts"
-
-
-/ {
-	regulators {
-		wlreg_on: fixedregulator@100 {
-			compatible = "regulator-fixed";
-			regulator-min-microvolt = <5000000>;
-			regulator-max-microvolt = <5000000>;
-			regulator-name = "wlreg_on";
-			gpio = <&gpio4 8 0>;
-			startup-delay-us = <100>;
-			enable-active-high;
-		};
-	};
-
-	bcmdhd_wlan_0: bcmdhd_wlan@0 {
-		compatible = "android,bcmdhd_wlan";
-		wlreg_on-supply = <&wlreg_on>;
-	};
-};
-
-
-&usdhc2 {
-	pinctrl-0 = <&pinctrl_usdhc2>;
-};
-
-
diff --git a/arch/arm/boot/dts/imx6sxscm-512mb-evb-btwifi-ldo.dts b/arch/arm/boot/dts/imx6sxscm-512mb-evb-btwifi-ldo.dts
new file mode 100644
index 0000000..6ef7aec
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-512mb-evb-btwifi-ldo.dts
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sxscm-512mb-evb-ldo.dts"
+#include "imx6sxscm-evb-btwifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6sxscm-512mb-evb-wifi-ldo.dts b/arch/arm/boot/dts/imx6sxscm-512mb-evb-wifi-ldo.dts
deleted file mode 100644
index 291e37f..0000000
--- a/arch/arm/boot/dts/imx6sxscm-512mb-evb-wifi-ldo.dts
+++ /dev/null
@@ -1,34 +0,0 @@
-/*
- * Copyright (C) 2016 Freescale Semiconductor, Inc.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-
-#include "imx6sxscm-512mb-evb-ldo.dts"
-
-/ {
-	regulators {
-		wlreg_on: fixedregulator@100 {
-			compatible = "regulator-fixed";
-			regulator-min-microvolt = <5000000>;
-			regulator-max-microvolt = <5000000>;
-			regulator-name = "wlreg_on";
-			gpio = <&gpio4 8 0>;
-			startup-delay-us = <100>;
-			enable-active-high;
-		};
-	};
-
-	bcmdhd_wlan_0: bcmdhd_wlan@0 {
-		compatible = "android,bcmdhd_wlan";
-		wlreg_on-supply = <&wlreg_on>;
-	};
-};
-
-
-&usdhc2 {
-	pinctrl-0 = <&pinctrl_usdhc2>;
-};
-
diff --git a/arch/arm/boot/dts/imx6sxscm-epop-evb-btwifi-ldo.dts b/arch/arm/boot/dts/imx6sxscm-epop-evb-btwifi-ldo.dts
new file mode 100644
index 0000000..0967a0f
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-epop-evb-btwifi-ldo.dts
@@ -0,0 +1,9 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+#include "imx6sxscm-512mb-evb-btwifi-ldo.dts"
+#include "imx6sxscm-emmc.dtsi"
diff --git a/arch/arm/boot/dts/imx6sxscm-epop-evb-wifi-ldo.dts b/arch/arm/boot/dts/imx6sxscm-epop-evb-wifi-ldo.dts
deleted file mode 100644
index 62bc1ff..0000000
--- a/arch/arm/boot/dts/imx6sxscm-epop-evb-wifi-ldo.dts
+++ /dev/null
@@ -1,9 +0,0 @@
-/*
- * Copyright (C) 2016 Freescale Semiconductor, Inc.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-#include "imx6sxscm-512mb-evb-ldo.dts"
-#include "imx6sxscm-emmc.dtsi"
diff --git a/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi b/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi
new file mode 100644
index 0000000..433b853
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/*
+ * NOTE: This DTS file is written for plugging in Murata Wi-Fi/BT EVK into SD3
+ * slot using Murata i.MX InterConnect Ver 2.0 Adapter AND ribbon cable connected
+ * to J15 on the EVB board. Bluetooth UART connects via ribbon cable located only 
+ * on Murata i.MX InterConnect ver2.0 adapter.
+ */
+
+
+/ {
+	regulators {
+		wlreg_on: fixedregulator@100 {
+			compatible = "regulator-fixed";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			regulator-name = "wlreg_on";
+			gpio = <&gpio4 8 0>;
+			startup-delay-us = <100>;
+			enable-active-high;
+		};
+		vcc_sd3: regulator@0 {
+			status = "disabled";
+		};
+
+	};
+
+	bcmdhd_wlan_0: bcmdhd_wlan@0 {
+		compatible = "android,bcmdhd_wlan";
+		wlreg_on-supply = <&wlreg_on>;
+	};
+};
+
+&iomuxc {
+	imx6sxscm-evb-murata-v2_rc {
+		pinctrl_bt: btgrp {
+			fsl,pins = <
+				MX6SX_PAD_NAND_DATA05__GPIO4_IO_9	0x13069 /* BT_REG_ON */
+			>;
+		};
+
+		pinctrl_uart6: uart6grp {
+			fsl,pins = <
+				MX6SX_PAD_KEY_COL1__UART6_TX            0x1b0b1
+				MX6SX_PAD_KEY_ROW1__UART6_RX            0x1b0b1
+				MX6SX_PAD_KEY_COL0__UART6_RTS_B		0x1b0b1
+				MX6SX_PAD_KEY_ROW0__UART6_CTS_B		0x1b0b1
+			>;
+		};
+
+		pinctrl_usdhc3_1: usdhc3grp-1 {
+			fsl,pins = <
+				MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x17069
+				MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x10071
+				MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x17069
+				MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x17069
+				MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x17069
+				MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x17069
+			>;
+		};
+
+		pinctrl_usdhc3_1_100mhz: usdhc3grp-1-100mhz {
+			fsl,pins = <
+				MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x170b9
+				MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x100b9
+				MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x170b9
+				MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x170b9
+				MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x170b9
+				MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x170b9
+			>;
+		};
+
+		pinctrl_usdhc3_1_200mhz: usdhc3grp-1-200mhz {
+			fsl,pins = <
+				MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x170f9
+				MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x100f9
+				MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x170f9
+				MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x170f9
+				MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x170f9
+				MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x170f9
+			>;
+		};
+
+		/* Murata change SD3 to 4-bit SDIO only; use upper 4-bits for UART. */
+		pinctrl_wifi: wifigrp {
+			fsl,pins = <
+				MX6SX_PAD_SD2_CMD__USDHC2_CMD		0x17069
+				MX6SX_PAD_SD2_CLK__USDHC2_CLK		0x10071
+				MX6SX_PAD_SD2_DATA0__USDHC2_DATA0	0x17069
+				MX6SX_PAD_SD2_DATA1__USDHC2_DATA1	0x17069
+				MX6SX_PAD_SD2_DATA2__USDHC2_DATA2	0x17069
+				MX6SX_PAD_SD2_DATA3__USDHC2_DATA3	0x17069
+				/* Murata Module control signals */ 
+				MX6SX_PAD_NAND_DATA04__GPIO4_IO_8	0x13069 /* WL_REG_ON */
+
+			>;
+		};
+	};
+};
+
+&uart6 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart6
+		     &pinctrl_bt>;
+	fsl,uart-has-rtscts;
+	status = "okay";
+};
+
+&usdhc2 {
+	pinctrl-0 = <&pinctrl_wifi>;
+	bus-width = <4>;
+};
+
+&usdhc3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc3_1>;
+	pinctrl-1 = <&pinctrl_usdhc3_1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc3_1_200mhz>;
+	bus-width = <4>;
+};
+
diff --git a/arch/arm/boot/dts/imx6sxscm-evb.dtsi b/arch/arm/boot/dts/imx6sxscm-evb.dtsi
index f628089..e52b8b0 100644
--- a/arch/arm/boot/dts/imx6sxscm-evb.dtsi
+++ b/arch/arm/boot/dts/imx6sxscm-evb.dtsi
@@ -246,3 +246,10 @@
 &flexcan2 {
     trx-en-gpio = <&gpio4 27 GPIO_ACTIVE_LOW>;
 };
+
+&usdhc3 {
+	cd-gpios = <>;
+	wp-gpios = <>;
+	vmmc-supply = <>;
+};
+
-- 
1.9.1

